### PR TITLE
Analysis Projects Require Environment Specifications

### PR DESCRIPTION
**Pull request by tmooney**
* Rename the "production" option in create to "prod-builder".
* Only allow an Analysis Project to be released if it has an environment configuration specified.

This PR doesn't enforce the contents of the environment configuration in any way.

### CODE REVIEW

File: lib/perl/Genome/Config/AnalysisProject/Command/Release.pm [hunk 1]
```diff
@@ -46,15 +46,27 @@ sub execute {
...
             push @errors, UR::Object::Tag->create(
                 type => 'error',
                 properties => ['analysis_projects'],
-                desc => "Can't release analysis project with status: $status"
+                desc => "Can't release analysis project " . $anp->__display_name__ . " with status: $status"
+            );
+        }
+
+        my $config = $anp->environment_config_dir;
+        unless ($config) {
```
Comment (acoffman) [inline lib/perl/Genome/Config/AnalysisProject/Command/Release.pm]: Should it maybe check that there's a non-empty file in there?

Comment (tmooney) [inline lib/perl/Genome/Config/AnalysisProject/Command/Release.pm]: The [`environment_config_dir`](https://github.com/tmooney/genome/blob/63fb747959461b19b699507aefc568bfdf8dcea3/lib/perl/Genome/Config/AnalysisProject.pm#L151) method does validate the path exists before returning it, but it doesn't check that there's really anything specified.

Would it be beneficial to add some extra checking there?  I considered parsing the environment in advance to look for particular keys, but it's not guaranteed that we'll need to use them for a particular build type.  We could at least look for some valid configuration file at all, though?

Comment (acoffman) [inline lib/perl/Genome/Config/AnalysisProject/Command/Release.pm]: Yeah, validating the contents of the config seems like a losing battle at this point. I was just debating seeing if there was a non-empty file of some type there, but if that's already being done in the accessor, I think its fine "as-is."

**Comment by acoffman** [comment]
+1

**Pull request merged**